eslint-rules/ts-no-use-before-define.js

```diff
@@ -0,0 +1,288 @@
+// Added here until https://github.com/nzakas/eslint-plugin-typescript/pull/34
```

Comment:
True, but that's specific to `var`s scope hoisting. If the linter ensures we're always using `let` or `const`, it'll catch it: